File: examples/typescript-vitest/steps/yahtzee.steps.ts
import { steps } from '@varar/varar'
import { score } from './yahtzee'

const { sensor } = steps()

// Header-bound table: the paragraph names every header cell (dice, category,
// score), so the runner calls this sensor once per row with the row as an
// object of raw strings. Returning { score } compares only that column.
sensor(
  'Examples of dice, category and score',
  (_state, row: { dice: string; category: string; score: string }) => {
    const dice = row.dice.split(',').map((d) => Number(d.trim()))
    return { score: score(dice, row.category) }
  },
)

File: examples/ruby-rspec/steps/yahtzee.steps.rb
# frozen_string_literal: true

require 'varar'
require_relative '../lib/yahtzee'

steps do
  # Header-bound table: the paragraph names every header cell (dice, category,
  # score), so this sensor runs once per row with the row as a hash keyed by
  # header. Returning {"score" => …} checks that column; the rest are inputs.
  sensor('Examples of dice, category and score') do |_state, row|
    dice = row['dice'].split(',').map { |d| d.strip.to_i }
    { 'score' => Yahtzee.score(dice, row['category']) }
  end
end
